server.go

```diff
@@ -412,6 +413,36 @@ func (c *serverConn) run(sctx context.Context) {
 		}
 	}(recvErr)
 
+	go func(responseErr chan error) {
```

Comment:
Should we be defer closing responseErr here?

Reply:
I don't think we want to close it, as that will cause the `case err := <-responseErr` branch to be hit on every iteration of the loop below. We should be okay to leave it open, it will be cleaned up by the GC once everything using it has stopped.

Reply:
Yeah, I don't think closing frees any resources, just makes it always hit.